### ISSUE TITLE
Is there shutdown/close methods of the client?

### ISSUE THREAD

**Issue by achimnol** [gaopeiliang/aioetcd3#5]
With my example code referred in #4 , I found that having many workers causes the following error when terminating:
```
Exception ignored when trying to write to the signal wakeup fd:
OSError: [Errno 9] Bad file descriptor
```
This doesn't happen when I skip all codes related to aiotecd3.
Maybe this is because there is no way to properly shutdown/close  the instantiated client object (with its underlyng grpc channel).

**Comment by gaopeiliang** [comment]
mybe manual `del client` last will clean up it !   
the example code https://gist.github.com/achimnol/b58810467db011a2f13323b3fab1ed9a , my Internet cannot open it!

**Comment by hubo1016** [comment]
aioetcd3 / aiogrpc uses grpc.io, which does not provide a close() method. It cleans up the resources on `__del__`. So remove the client and let gc collects it should do the work.

**Comment by achimnol** [comment]
Adding explicit `del` after use does not provide full clean & graceful shutdown, particularly when there are watchers.
At least I no longer see the above OSError, but still see a lot of aiogrpc's screams :scream: about "RuntimeError: Event loop is closed!".

**Comment by achimnol** [comment]
Also I'm curious about what if new watchers are accumulated over time in the etcd server.
Does the etcd3 server automatically clean up watchers registered by disconnected clients?

**Comment by hubo1016** [comment]
Seems that we have to trace all on-calling methods and cancel them for an immediately shutdown, or wait for all the methods to be finished. Closing the event loop should be done after that.

For etcd3, there are only two streaming APIs: watch and keepalive. So maybe we can just close the watch stream and wait for other methods to be completed (they usually finish very quickly, or at least have a 5-seconds-deadline by default).

The upstream project `aiogrpc` must be updated for this. I will do it when I have time.

**Comment by hubo1016** [comment]
@achimnol Watchers should be cleaned up when disconnected.

**Comment by hubo1016** [comment]
grpcio 1.12.0 introduces the close method, so this can be solved now

**Comment by hubo1016** [comment]
Fixed with #13
